addons/website_form/static/src/snippets/s_website_form/000.js

```diff
@@ -192,17 +193,31 @@ odoo.define('website_form.s_website_form', function (require) {
                         successMode = successPage ? 'redirect' : 'nothing';
                     }
                     switch (successMode) {
-                        case 'redirect':
-                            successPage = successPage.startsWith("/#") ? successPage.slice(1) : successPage;
+                        case 'redirect': {
+                            const hashIndex = successPage.indexOf("#");
+                            if (hashIndex > 0) {
+                                // Compare the redirect link (with and without
+                                // the lang url code) with the current page to
+                                // know if if we should stay on the same page
+                                // in the case of an anchor link.
+                                const target = successPage.replaceAll("/", "");
+                                if ([target, session.lang_url_code + target].some(link => link.startsWith(window.location.pathname.replaceAll("/", "") + '#'))) {
+                                    successPage = successPage.substring(hashIndex);
```

Comment:
because for instance `session.lang_url_code + successPage` could be `fr_BE/contactus#ok%C3%A9%25`
when `window.location.pathname  + '#'` is `/fr_BE/contactus`

Reply:
Ok, I got confused by the `replaceAll` which is removing all the slashes not only the first one that you seem to need.
If you create an util method, all that logic could be encapsulated inside it with small comments, that would be easier IMHO.
What do you think?

On top of the method:
```js
// Ensure no leading slash on both url for comparison (or the opposite, add it if not)
url1 = / + url1 if not url1 starts with /
url2 = / + url2 if not url2 starts with /
```

Reply:
Actually, it's not only about "start with '/' ", `window.location.pathname` could also end by "/" or not... 
My logic was: "let's get rid of all the "/" to make the comparison.
But ok, I changed it to be more lisible...

Reply:
In the other commit, I would have to do something like this to avoid the replaceAll().. which do you prefer?
```
        let targetPath = this.$target[0].pathname;
        let currentUrlPath = window.location.pathname;
        if (!targetPath.startsWith("/")) {
            targetPath = "/" + targetPath;
        }
        if (!currentUrlPath.endsWith("/")) {
            currentUrlPath = currentUrlPath + "/";
        }
        if (targetPath !== currentUrlPath) {
            return;
        }
```